cpp/perspective/CMakeLists.txt

```diff
@@ -131,7 +131,7 @@ elseif(PSP_PYTHON_BUILD)
     if(NOT DEFINED PSP_PYTHON_VERSION)
```

Comment:
This file seems to be getting messier and messier, and this PR seems to exacerbate that by making the control flow even more complicated

Reply:
Agreed this file could use a makeover. I'm not sure where `PSP_WASM_BUILD` and `PSP_PYTHON_BUILD` are being conflated though. This is my understanding, correct me if I'm wrong:
* `PSP_WASM_BUILD` only is a JS target.
* `PSP_PYTHON_BUILD` only is a Python native dylib.
* `PSP_WASM_BUILD` and `PSP_PYTHON_BUILD` together means we're building an emscripten SIDE_MODULE loadable by any compatible Python implementation.

If there's somewhere this is being handled incorrectly, we definitely need to fix it. I'll fix the `MANYLINUX` thing when I'm at my computer again, thanks for catching that 👍

Reply:
depending on the level of overlap, we could just have 3 completely separate paths. I dont think theres much overlap between python and pyodide, since we’re not going to support building from pyodide on e.g. windows. Emscripten and pyodide of some overlapping flags, but if theres enough separation we can literally just do:

```raw
if(PSP_WASM_BUILD)
set(PSP_PYTHON_BUILD OFF)
set(PSP_PYODIDE_BUILD OFF)
…
elseif(PSP_PYODIDE_BUILD)
…
elseif(PSP_PYTHON_BUILD)
…
endif()
```